src/sql/workbench/browser/ui/infoBox/infoBox.ts

```diff
@@ -98,14 +113,89 @@ export class InfoBox extends Disposable implements IThemable {
 		this.updateStyle();
 	}
 
+	public get links(): azdata.LinkArea[] {
+		return this._links;
+	}
+
+	public set links(v: azdata.LinkArea[]) {
+		this._links = v ?? [];
+	}
+
 	public get text(): string {
 		return this._text;
 	}
 
 	public set text(text: string) {
 		if (this._text !== text) {
 			this._text = text;
-			this._textElement.innerText = text;
+
+			DOM.clearNode(this._textElement);
+			this._linkListenersDisposableStore.clear();
+
+			for (let i = 0; i < this._links.length; i++) {
+				const placeholderIndex = text.indexOf(`{${i}}`);
+				if (placeholderIndex < 0) {
+					this._logService.warn(`Could not find placeholder text {${i}} in text ${text}`);
+					// Just continue on so we at least show the rest of the text if just one was missed or something
+					continue;
+				}
+
+				// First insert any text from the start of the current string fragment up to the placeholder
+				let curText = text.slice(0, placeholderIndex);
+				if (curText) {
+					const span = DOM.$('span');
+					span.innerText = text.slice(0, placeholderIndex);
+					this._textElement.appendChild(span);
+				}
+
+				// Now insert the link element
+				const link = this._links[i];
+				const linkElement = DOM.$('a', {
+					href: link.url === '' ? ' ' : link.url
+				});
+
+				linkElement.innerText = link.text;
+
+				if (link.accessibilityInformation) {
+					linkElement.setAttribute('aria-label', link.accessibilityInformation.label);
+					if (link.accessibilityInformation.role) {
+						linkElement.setAttribute('role', link.accessibilityInformation.role);
+					}
+				}
+				this._linkListenersDisposableStore.add(DOM.addDisposableListener(linkElement, DOM.EventType.CLICK, e => {
+					this._onLinkClick.fire({
+						index: i,
+						link: link
+					});
+					if (link.url) {
+						this.openLink(link.url);
+					}
+					e.stopPropagation();
+				}));
+
+				this._linkListenersDisposableStore.add(DOM.addDisposableListener(linkElement, DOM.EventType.KEY_PRESS, e => {
+					const event = new StandardKeyboardEvent(e);
+					if (this._isClickable && (event.equals(KeyCode.Enter) || !event.equals(KeyCode.Space))) {
+						this._onLinkClick.fire({
+							index: i,
+							link: link
+						});
+						if (link.url) {
+							this.openLink(link.url);
+						}
+						e.stopPropagation();
+					}
+				}));
+				this._textElement.appendChild(linkElement);
+				text = text.slice(placeholderIndex + 3);
+			}
+
+			if (text && typeof text === 'string') {
```

Comment:
You can just do `typeof text === 'string'`, if it's undefined that'll be false

But what is this check even supposed to be doing?